marker_driver_rustc/src/conversion/marker/expr.rs

```diff
@@ -314,4 +339,55 @@ impl<'ast, 'tcx> MarkerConverterInner<'ast, 'tcx> {
         let data = CommonExprData::new(self.to_expr_id(lets.hir_id), self.to_span_id(lets.span));
         ExprKind::Let(self.alloc(LetExpr::new(data, self.to_pat(lets.pat), self.to_expr(lets.init))))
     }
+
+    /// Rustc desugars assignments with tuples, arrays and structs in the
+    /// assignee as a `let` statements. The pattern than assign the values
+    /// to temporary variables called `lhs` which are then assigned to the
+    /// target value.
+    ///
+    /// The "Show HIR" option on the [Playground] is a great resource to
+    /// understand how this desugaring works. Here is a simple example to
+    /// illustrate the current desugar:
+    ///
+    /// ```
+    /// # let mut a = 0;
+    /// # let mut b = 0;
+    /// // This expression
+    /// [a, b] = [1, 2];
+    /// // Is desugared to, note that both lhs have different IDs
+    /// { let [lhs, lhs] = [1, 2]; a = lhs; b = lhs; };
+    /// ```
+    ///
+    /// [Playground]: https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021
+    fn to_assign_expr_from_desugar(&self, block: &hir::Block<'tcx>) -> AssignExpr<'ast> {
+        let lhs_map: FxHashMap<_, _> = block
+            .stmts
+            .iter()
+            .skip(1)
+            .map(|stmt| {
+                if let hir::StmtKind::Expr(expr) = stmt.kind
+                    && let hir::ExprKind::Assign(ass, value, _span) = expr.kind
```

Comment:
This looks like a pretty unfortunate variable name :sweat_smile:

Maybe something like `assign_expr`?

Reply:
Ups, changing this was on my todo list and then I forgot ^^